File: 프로그래머스/1/72410. 신규 아이디 추천/신규 아이디 추천.py
def solution(new_id):
    answer = ''
    canuse = {"-","_","."}
    pw = new_id.lower() #1단계
    pw = "".join(p for p in pw if p in canuse or p.isalnum()) #2단계
    while ".." in pw: #3단계
        pw = pw.replace("..",".")
    pw = pw.strip(".") #4단계
    pw = pw if pw else "a" #5단계
    if len(pw) >= 16: #6단계
        pw = pw[:15]
    pw = pw.rstrip(".")
    while len(pw) < 3: #7단계
        pw += pw[-1]  
    return pw

File: 프로그래머스/1/72410. 신규 아이디 추천/신규 아이디 추천.cpp
#include <string>
#include <vector>
#include <algorithm>
#include <iostream>
#include <cctype>

using namespace std;

string solution(string new_id) {
    string answer = "";
    transform(new_id.begin(), new_id.end(), new_id.begin(), ::tolower); // 1
    
    for(auto x: new_id) { // 2
        if (isalnum(x) || x == '-' || x == '_' || x == '.') {
            answer += x;
        } else continue;
    }
    
    char prev = answer[0];
    string tmp = "";
    tmp += prev;
    for (int i=1; i<answer.length();i++) {
        if (answer[i] == '.') {
            if (prev == '.') {
                continue;
            } else {
                tmp += answer[i];
                prev = '.';
            }
        } else {
            tmp += answer[i];
            prev = ' ';
        }
    }
    answer = tmp; // 3
    
    if (answer.front() == '.') {
        answer.erase(0,1);
    }
    if (answer.back() == '.') {
        answer.erase(answer.length()-1,1);
    } // 4
    
    if (answer.empty()) answer = 'a'; // 5
    if (answer.length() >= 16) {
        answer.erase(15, answer.length()-15);
    }
    if (answer.back() == '.') {
        answer.erase(answer.length()-1,1);
    } // 6
    if (answer.length() <= 2) {
        char last = answer.back();
        while (answer.length() < 3) {
            answer += last;
        }
    }
    
    return answer;
}